tests/test_core.py

```diff
@@ -8,15 +10,52 @@ class test_Interpolator(unittest.TestCase):
 
     def test_init(self):
         """Test creation from data."""
-        raise NotImplementedError()
+
+        #test construction with tuple input
+        f = relentless.core.Interpolator(x=(-1,0,1), y=(-2,0,2))
+        self.assertEqual(f._domain, (-1,1))
+        self.assertIsInstance(f._spline, Akima1DInterpolator)
+
+        #test construction with list input
+        f = relentless.core.Interpolator(x=[-1,0,1], y=[-2,0,2])
+        self.assertEqual(f._domain, (-1,1))
+        self.assertIsInstance(f._spline, Akima1DInterpolator)
+
+        #test construction with numpy array input
+        f = relentless.core.Interpolator(x=np.array([-1,0,1]), y=np.array([-2,0,2]))
+        self.assertEqual(f._domain, (-1,1))
+        self.assertIsInstance(f._spline, Akima1DInterpolator)
+
+        #test construction with non-increasing domain
+        with self.assertRaises(ValueError):
+            f = relentless.core.Interpolator(x=(1,0,-1), y=(2,0,-2))
```

Comment:
Good idea. Are there other tests that might fail? What about if x and y are only scalars? (Not actually sure what happens there. If it works, then no need to test this.)